src/channelimpl.cpp

```diff
@@ -290,6 +290,7 @@ Deferred &ChannelImpl::declareExchange(const std::string &name, ExchangeType typ
     else if (type == ExchangeType::topic)           exchangeType = "topic";
     else if (type == ExchangeType::headers)         exchangeType = "headers";
     else if (type == ExchangeType::consistent_hash) exchangeType = "x-consistent-hash";
+    else if (type == ExchangeType::consistent_hash) exchangeType = "x-message-deduplication";
```

Comment:
this should be `message_deduplication` instead of `consistent_hash`?

Reply:
yes indeed really sorry I fix right now

Reply:
I [sent a PR](https://github.com/CopernicaMarketingSoftware/AMQP-CPP/pull/438) with the fix :/